src/core/scene/a-scene.js

```diff
@@ -123,37 +143,64 @@ module.exports = registerElement('a-scene', {
       }
     },
 
-    /**
-     * Generally must be triggered on user action for requesting fullscreen.
-     */
     enterVR: {
       value: function (event) {
         var self = this;
-        return this.effect.requestPresent().then(enterVRSuccess, enterVRFailure);
+        if (this.is('vr-mode')) { return; }
+        if (checkHeadsetConnected() || this.isMobile) {
+          // We send the canvas fullscreen so mirroring overlays
+          // any other elements on the page. Important with
+          // embedded scenes.
+          if (!this.isMobile) { this.requestFullscreen(); }
+          return this.effect.requestPresent().then(enterVRSuccess, enterVRFailure);
+        }
+        // Just go fullscreen with no headset
+        this.requestFullscreen();
+        enterVRSuccess();
         function enterVRSuccess () {
           self.addState('vr-mode');
           self.emit('enter-vr', event);
           // Lock to landscape orientation on mobile.
           if (self.isMobile && window.screen.orientation) {
             window.screen.orientation.lock('landscape');
           }
+          self.addFullScreenStyles();
+          self.resize();
         }
         function enterVRFailure () {
           throw new Error('enter VR mode error. requestPresent failed');
         }
       }
     },
 
+    requestFullscreen: {
+      value: function () {
+        var canvas = this.canvas;
+        var requestFullscreen =
+          canvas.requestFullScreen ||
+          canvas.webkitRequestFullScreen ||
+          canvas.mozRequestFullScreen;
+        requestFullscreen.apply(canvas);
+      }
+    },
+
     exitVR: {
       value: function () {
         var self = this;
-        return this.effect.exitPresent().then(exitVRSuccess, exitVRFailure);
+        if (!this.is('vr-mode')) { return; }
```

Comment:
it's more readable overall because you do all your checks at once and there's a logical progression to the function

1. Check if I can exit VR
2. Exit VR 

Rather than

1. Check if I'm in VR to see if I can exit VR 
2. If I can exit VR, exit VR